packages/deployments/contracts/contracts/core/connext/facets/BaseConnextFacet.sol

```diff
@@ -168,4 +171,35 @@ contract BaseConnextFacet {
   function _originAndNonce(uint32 _origin, uint32 _nonce) internal pure returns (uint64) {
     return (uint64(_origin) << 32) | _nonce;
   }
+
+  function _getLocalAsset(bytes32 _id, uint32 _domain) internal view returns (address _token) {
+    _token = AssetLogic.getLocalAsset(_id, _domain, s);
+  }
+
+  function _getCanonicalTokenId(address _candidate) internal view returns (TokenId memory) {
+    return AssetLogic.getCanonicalTokenId(_candidate, s);
+  }
+
+  function _getLocalAndAdoptedToken(bytes32 _id, uint32 _domain)
+    internal
+    view
+    returns (address _local, address _adopted)
+  {
+    _local = AssetLogic.getLocalAsset(_id, _domain, s);
+    _adopted = _getAdoptedAsset(_id, _domain);
+  }
+
+  function _isLocalOrigin(address _token) internal view returns (bool) {
+    return AssetLogic.isLocalOrigin(_token, s);
+  }
+
+  function _getApprovedCanonicalId(address _candidate) internal view returns (TokenId memory _canonical) {
```

Comment:
functions like this are really useful if we're calling another contract - they do a bunch of needed things at once.

however, in what is essentially a library (an inherited contract), you may as well have two separate functions - one for getting the canonical token ID and the other for checking whether the asset is approved. all the logic is effectively in-lined anyway, so we might as well just make this function for checking whether the canonical ID is approved, and enforce that the caller has already called `_getCanonicalTokenId` separately

Reply:
tldr IMO avoid over-specialized helpers idk. this is a nit lol

Reply:
agree but really its to share the helper between two facets -- the code path would be the same so i think putting them in the same helper is okay